### DISCUSSION TITLE
Add monaco plugins to CPL

### DISCUSSION THREAD

**Discussion by Gk0Wk** [SmilyOrg/tw5-monaco#10] [Show and tell]
I have added the [Monaco plugin](https://tw-cpl.netlify.app/#Plugin_202203243927762:Plugin_202203243927762%20Plugin_202203243930466) to the CPL plugin library (which, of course, automatically keeps the latest version), and now users can install and update the Monaco plugin more easily through CPL.

[What is CPL](https://talk.tiddlywiki.org/t/tiddlywiki-cpl-tiddlywiki-world-of-google-app-store/2140)

**Comment by SmilyOrg** [comment]
Awesome, thank you! How does it work technically? Does it pull from GitHub or GH Pages?

**Comment by Gk0Wk** [comment]
This is roughly how it works: add a plugin-info tiddler (containing title, download URI, etc.) to the CPL, download files from these URI, and try to import the corresponding plugin tiddler each time a build is performed, then generate the plugin library.

![CPL](https://user-images.githubusercontent.com/16955102/160115724-1495eca0-ae83-41f3-bd95-764ac3602761.png)

[View the build script](https://github.com/tiddly-gittly/TiddlyWiki-CPL/blob/master/scripts/index.js)

**Reply by SmilyOrg** [reply]
Cool stuff! I see it's sourced from GitHub pages, neat :)

**Comment by SmilyOrg** [comment]
Just tried it out and it seems to work pretty well. I like the comments and reactions integration, how did you set that up?

I also noticed that monaco throws an error if you don't also install monaco-wikitext right now, but it also happens with other installation methods, so it's not because of CPL. Not sure if there's a lot of benefit in having that as separate plugins anyway 🤔 (created #12 for this)
